src/backend/commands/tablecmds.c

```diff
@@ -7604,6 +7601,21 @@ ATExecAddColumn(List **wqueue, AlteredTableInfo *tab, Relation rel,
 	if (!HeapTupleIsValid(reltup))
 		elog(ERROR, "cache lookup failed for relation %u", myrelid);
 	relkind = ((Form_pg_class) GETSTRUCT(reltup))->relkind;
+	relstorage = ((Form_pg_class) GETSTRUCT(reltup))->relstorage;
+
+	/* Check for NOT NULL constraints in external readable tables */
+	if (colDef->is_not_null && relkind == RELKIND_RELATION && relstorage == RELSTORAGE_EXTERNAL)
```

Comment:
Could we use RelationIsExternal(rel)  as below?

Reply:
No, we don't have Relation object yet, only the separate relkind and relstorage

Reply:
Sorry, don't quite understand it. There is a parameter Relation rel, which has 
```
p rel->rd_rel->relstorage
$1 = 120 'x'
```

Reply:
Sorry, I thought this was referring to something else. Yes, here it is possible to use RelationIsExternal, I will replace it